oxd-server/src/main/java/org/gluu/oxd/server/service/HttpService.java

```diff
@@ -34,26 +36,39 @@ public HttpService(OxdServerConfiguration configuration) {
     }
 
     public HttpClient getHttpClient() {
+        final ProxyConfiguration proxyConfig = asProxyConfiguration(configuration);
         try {
             final Boolean trustAllCerts = configuration.getTrustAllCerts();
             if (trustAllCerts != null && trustAllCerts) {
                 LOG.trace("Created TRUST_ALL client.");
-                return CoreUtils.createHttpClientTrustAll();
+                return CoreUtils.createHttpClientTrustAll(proxyConfig);
             }
             final String keyStorePath = configuration.getKeyStorePath();
             if (StringUtils.isNotBlank(keyStorePath)) {
                 final File keyStoreFile = new File(keyStorePath);
                 if (!keyStoreFile.exists()) {
                     LOG.error("ERROR in configuration. Key store path is invalid! Please fix key_store_path in oxd configuration");
                 } else {
-                    return CoreUtils.createHttpClientWithKeyStore(keyStoreFile, configuration.getKeyStorePassword());
+                    return CoreUtils.createHttpClientWithKeyStore(keyStoreFile, configuration.getKeyStorePassword(), proxyConfig);
                 }
             }
         } catch (Exception e) {
             LOG.error(e.getMessage(), e);
             LOG.error("Failed to create http client based on oxd configuration. Created default client.");
         }
-        return new DefaultHttpClient();
+        return CoreUtils.createClient(null, proxyConfig);
+    }
+
+    private static ProxyConfiguration asProxyConfiguration(OxdServerConfiguration configuration) {
+        try {
+            JsonNode node = configuration.getProxyConfiguration();
+            if (node != null) {
+                return Jackson2.createJsonMapper().treeToValue(node, ProxyConfiguration.class);
+            }
+        } catch (Exception e) {
+            LOG.error("Failed to parse ProxyConfiguration.", e);
+        }
+        return new ProxyConfiguration();
```

Comment:
return null and handle it. Or otherwise `Optional<ProxyConfiguration>`.

Reply:
Done. Used `Optional<ProxyConfiguration>`